### PR TITLE
LibWeb+LibIDL: Add support for overloading constructors

### PR DESCRIPTION
**Pull request by kennethmyhra**
This enables our bindings generator to properly handle overloaded constructors.

This also adds Uint8ClampedArray as supported parameter type to our BindingsGenerator + adds the second ImageData constructor which takes an array of Uint8ClampedArray as data.

### CODE REVIEW

**Comment by kennethmyhra** [comment]
Red CI seems related to general CI flakiness

**Comment by Lubrsi** [comment]
Red CI unrelated:
```
Timeout: /home/vsts/work/1/s/Tests/LibWeb/Text/input/WebAnimations/misc/animation-single-iteration-no-repeat.html
```
```
Fail: /Users/runner/work/1/s/Tests/LibWeb/Text/input/WebAnimations/animation-properties/replaceState.html
```
```diff
--- /Users/runner/work/1/s/Tests/LibWeb/Text/expected/./WebAnimations/animation-properties/replaceState.txt
+++ /Users/runner/work/1/s/Tests/LibWeb/Text/expected/./WebAnimations/animation-properties/replaceState.txt
@@ -1,4 +1,4 @@
    Animation's replaceState is active initially: true
 Animation's replaceState is active after finishing: true
 Animation's replaceState is not removed after creating new animation: true
-Animation's replaceState is removed after new animation finishes: true
+Animation's replaceState is removed after new animation finishes: false
```

File: Userland/Libraries/LibWeb/HTML/ImageData.cpp [hunk 1]
```diff
@@ -37,6 +39,50 @@ WebIDL::ExceptionOr<JS::NonnullGCPtr<ImageData>> ImageData::construct_impl(JS::R
...
+    if (!is<JS::Uint8ClampedArray>(*data->raw_object()))
+        return vm.throw_completion<JS::TypeError>(JS::ErrorType::NotAnObjectOfType, "Uint8ClampedArray");
+
+    auto& uint8_clamped_array_data = static_cast<JS::Uint8ClampedArray&>(*data->raw_object());
+
+    // 1. Let length be the number of bytes in data.
+    auto length = uint8_clamped_array_data.byte_length().length();
+
+    // 2. If length is not a nonzero integral multiple of four, then throw an "InvalidStateError" DOMException.
+    if (length % 4 != 0)
```
Comment (Lubrsi) [inline Userland/Libraries/LibWeb/HTML/ImageData.cpp:56]: I didn't notice this on first read and thought the NOTE on step 4 didn't make sense, but this doesn't quite match the text as it's missing the `nonzero` part, the integral multiple of four is correct though.

Comment (kennethmyhra) [reply Userland/Libraries/LibWeb/HTML/ImageData.cpp]: Great catch!

**Comment by kennethmyhra** [comment]
Rebased after flaky tests were disabled to get a green CI run.

**Pull request merged**